### PR TITLE
Update disk cache documentation

### LINKED ISSUE #5739: Improve disk-caching document based on feedback from Slack
[ Not using Issue template since it's not useful in this case ]

- [x] Clarify usage of terms like cache-directories v/s cache-drives (Usage of drives suggests that it doesn't require a filesystem on top)
- [x] What happens to data in cache directories when minio cache server is restarted?

Thanks, @vRobM for suggesting these improvements.

### PR DESCRIPTION
**Pull request by poornas**
fixes #5739 
## How Has This Been Tested?

manually
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/disk-caching/DESIGN.md [hunk 1]
```diff
@@ -42,6 +43,10 @@ Disk caching caches objects for both **uploaded** and **downloaded** objects i.e
 
 > NOTE: Expiration happens automatically based on the configured interval as explained above, frequently accessed objects stay alive in cache for a significantly longer time.
 
+### Crash Recovery
+Upon restart of minio server after a running minio process is killed or crashes, disk caching resumes automatically. The garbage collection cycle resumes and any previously
+cached entries are served from cache.
```
Comment (harshavardhana) [inline docs/disk-caching/DESIGN.md]: don't manually press enter let markdown to wrap itself.

File: docs/disk-caching/DESIGN.md [hunk 2]
```diff
@@ -9,24 +9,25 @@ minio server -h
...
      $ export MINIO_CACHE_EXCLUDE="mybucket/*;*.pdf"
      $ export MINIO_CACHE_EXPIRY=40
      $ minio server /home/shared
 ```
 
 ## Assumptions
 - Disk cache size defaults to 80% of your drive capacity.
-- Disk caching requires [`atime`](http://kerolasa.github.io/filetimes.html) support to be enabled on the cache drive.
+- The cache drives are required to be a filesystem mount point with [`atime`](http://kerolasa.github.io/filetimes.html) support to be enabled on the drive. Alternatively writable directories 
+  with atime support can be specified in MINIO_CACHE_DRIVES
```
Comment (harshavardhana) [inline docs/disk-caching/DESIGN.md]: don't manually press enter let markdown to wrap itself.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

File: docs/disk-caching/README.md [hunk 3]
```diff
@@ -11,20 +11,20 @@ Disk caching feature here refers to the use of caching disks to store content cl
 Install Minio - [Minio Quickstart Guide](https://docs.minio.io/docs/minio).
 
 ### 2. Run Minio with cache
-Disk caching can be enabled by updating the `cache` config settings for Minio server. Config `cache` settings takes the drive locations, cache expiry duration (in days) and any wildcard patterns to exclude from being cached.
+Disk caching can be enabled by updating the `cache` config settings for Minio server. Config `cache` settings takes the mounted drive(s) or director(ies) paths, cache expiry duration (in days) and any wildcard patterns to exclude from being cached.
```
Comment (nitisht) [inline docs/disk-caching/README.md]: We can just say directory path(s) instead of `director(ies) paths`

**Review by nitisht — CHANGES_REQUESTED** [review]

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Review by nitisht — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**